pkg/controller/common/helpers.go

```diff
@@ -740,3 +743,50 @@ func GetIgnitionFileDataByPath(config *ign3types.Config, path string) ([]byte, e
 	}
 	return nil, nil
 }
+
+// GetNewestCertificatesFromPEMBundle breaks a pem-encoded bundle out into its component certificates
+func GetCertificatesFromPEMBundle(pemBytes []byte) ([]*x509.Certificate, error) {
+	var certs []*x509.Certificate
+	// There can be multiple certificates in the file
+	for {
```

Comment:
wait sorry - do we really want an infinite loop here? could we make this a little more explicit?

Reply:
Future John will come back to this later to add clarifying comments which is fine with me.